### PR TITLE
Rewrite changelog for beta4 in light of recent developments

### PR DESCRIPTION
**Pull request by glibsm**
It's quite difficult to capture the magnitude of changes, but I tried to do my best.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,30 +1,21 @@
...
-  explicit constructors `WithExpand` to use interpolation. Only values in the YAML
-  files are replaced, comments and keys are ignored.
-- Add a SetStaticConfigFiles method to load yaml files that should not be interpolated.
+- **[Breaking]** Monolithic framework, as released in initial betas, has been
+  broken into smaller pieces as a result of recent advances in `dig` library.
+  This is a radical departure from the previous direction, but it needed to
+  be done for the long-term good of the project.
+- **[Breaking]** `Module interface` has been scoped all the way down to being
+  *a single dig constructor*. This allows for very sophisticated module
+  compositions. See `uber-go/dig` for more information on the constructors.
```
Comment (abhinav) [inline CHANGELOG.md]: nit: mention go.uber.org/dig instead

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,30 +1,21 @@
...
+- **[Breaking]** `package config` has been moved to `github.com/uber-go/config`,
+  importable via `go.uber.org/config`.
```
Comment (abhinav) [inline CHANGELOG.md]: nit: mention just `go.uber.org/config`. finding the repo from the docs is easy plus the docs are really the thing most users want to look at.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,30 +1,21 @@
...
+- `fx.Lifecycle` has been added for modules to hook into the framework
+  lifecycle events.
+- `service.Host` interface which composed a number of primitives together
+  (configuration, metrics, tracing) has been deprecated with the top-level
+  construct renamed to `fx.App`.
```
Comment (abhinav) [inline CHANGELOG.md]: Can shorten to,

> [..] deprecated in favor of `fx.App`.

**Comment by glibsm** [comment]
@alsamylkin config is not imported directly from Fx, but rather by the stacks/services. So it's OK that it's not released just yet.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,30 +1,21 @@
...
-  files are replaced, comments and keys are ignored.
-- Add a SetStaticConfigFiles method to load yaml files that should not be interpolated.
+- **[Breaking]** Monolithic framework, as released in initial betas, has been
+  broken into smaller pieces as a result of recent advances in `dig` library.
+  This is a radical departure from the previous direction, but it needed to
+  be done for the long-term good of the project.
+- **[Breaking]** `Module interface` has been scoped all the way down to being
+  *a single dig constructor*. This allows for very sophisticated module
+  compositions. See `go.uber.org/dig` for more information on the constructors.
+- **[Breaking]** `package config` has been moved to it's own repository.
```
Comment (abhinav) [inline CHANGELOG.md]: > *its* own repository.

**Pull request merged**